test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java

```diff
@@ -561,6 +562,20 @@ public void indexCorrectlyMarkedAsBuildAndRemoved() throws Throwable
         String selectBuiltIndexesQuery = String.format("SELECT * FROM %s.\"%s\"",
                                                        SchemaConstants.SYSTEM_KEYSPACE_NAME,
                                                        SystemKeyspace.BUILT_INDEXES);
+
+        // Wait for any background index clearing tasks to complete
+        Util.spinAssertEquals(0, () -> {
+            try
+            {
+                return execute(selectBuiltIndexesQuery).size();
+            }
+            catch(Throwable e)
+            {
+                fail(e.getMessage());
+            }
+            return -1;
+        }, 60);
```

Comment:
We can use the briefer and more flexible `Awaitility.await()` method, that doesn't require the try-catch block and allows us to specify a poll delay:
```suggestion
        await().atMost(1, MINUTES)
               .pollDelay(1, SECONDS)
               .untilAsserted(() -> assertRows(execute(selectBuiltIndexesQuery)));
```

Reply:
So what's the point of having 2 methods that do the same? Is it bc one uses Thread.yield()?

Reply:
I don't know, I think the dependency for the more sophisticated `Awaitability` was added quite recently by CASSANDRA-15677. Maybe it was because `spinAssertEquals` is not visible from dtests? Perhaps we should consider getting rid of one of them in a separate ticket?

Reply:
That's what I was thinking. Unless Thread.yield() provides some advantage bc it's more aggressive we could just replace that method contents with Awaitility...

Reply:
https://issues.apache.org/jira/browse/CASSANDRA-16621